server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -287,12 +284,13 @@ export function configureWebSocketServices(
                 });
 
             if (clients.length > maxNumberOfClientsPerDocument) {
-                // eslint-disable-next-line prefer-promise-reject-errors
-                return Promise.reject({
-                    code: 429,
-                    message: "Too Many Clients Connected to Document",
-                    retryAfter: 5 * 60,
-                });
+                throw new NetworkError(
+                    429,
```

Comment:
nit: you may consider adding comments against these params to help with readability.